privacy-protections/fingerprinting/helpers/tests.js

```diff
@@ -1018,62 +1034,66 @@ const tests = [
             canvas.width = 2000
             canvas.height = 200
             canvas.style.display = 'inline'
+
+            applyFpExampleDataToCanvas(canvas)
+            addCanvasToPage(canvas, 'Normal check')
+
             const ctx = canvas.getContext('2d')
-            // detect browser support of canvas winding
-            // http://blogs.adobe.com/webplatform/2013/01/30/winding-rules-in-canvas/
-            // https://github.com/Modernizr/Modernizr/blob/master/feature-detects/canvas/winding.js
-            ctx.rect(0, 0, 10, 10)
-            ctx.rect(2, 2, 6, 6)
-        
-            ctx.textBaseline = 'alphabetic'
-            ctx.fillStyle = '#f60'
-            ctx.fillRect(125, 1, 62, 20)
-            ctx.fillStyle = '#069'
-            // https://github.com/Valve/fingerprintjs2/issues/66
-            // if (this.options.dontUseFakeFontInCanvas) {
-            // ctx.font = '11pt Arial'
-            // } else {
-            ctx.font = '11pt no-real-font-123'
-            // }
-            ctx.fillText('Cwm fjordbank glyphs vext quiz, \ud83d\ude03', 2, 15)
-            ctx.fillStyle = 'rgba(102, 204, 0, 0.2)'
-            ctx.font = '18pt Arial'
-            ctx.fillText('Cwm fjordbank glyphs vext quiz, \ud83d\ude03', 4, 45)
-        
-            // canvas blending
-            // http://blogs.adobe.com/webplatform/2013/01/28/blending-features-in-canvas/
-            // http://jsfiddle.net/NDYV8/16/
-            ctx.globalCompositeOperation = 'multiply'
-            ctx.fillStyle = 'rgb(255,0,255)'
-            ctx.beginPath()
-            ctx.arc(50, 50, 50, 0, Math.PI * 2, true)
-            ctx.closePath()
-            ctx.fill()
-            ctx.fillStyle = 'rgb(0,255,255)'
-            ctx.beginPath()
-            ctx.arc(100, 50, 50, 0, Math.PI * 2, true)
-            ctx.closePath()
-            ctx.fill()
-            ctx.fillStyle = 'rgb(255,255,0)'
-            ctx.beginPath()
-            ctx.arc(75, 100, 50, 0, Math.PI * 2, true)
-            ctx.closePath()
-            ctx.fill()
-            ctx.fillStyle = 'rgb(255,0,255)'
-            // canvas winding
-            // http://blogs.adobe.com/webplatform/2013/01/30/winding-rules-in-canvas/
-            // http://jsfiddle.net/NDYV8/19/
-            ctx.arc(75, 75, 75, 0, Math.PI * 2, true)
-            ctx.arc(75, 75, 25, 0, Math.PI * 2, true)
-            ctx.fill('evenodd')
+            const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height)
+
+            // Verify that once the method is used that it looks the same once rendered to the screen
+            const canvasCopy = document.createElement('canvas')
+            canvasCopy.width = 2000
+            canvasCopy.height = 200
+            canvasCopy.style.display = 'inline'
+            const ctxCopy = canvasCopy.getContext('2d')
+            ctxCopy.putImageData(imageData, 0, 0)
+            addCanvasToPage(canvasCopy, 'Image data check')
```

Comment:
should we have one more of those for toDataURL?

Reply:
I've added this as an image, I was mostly trying to avoid then having to take the data from an image then have to turn it into a canvas as it wouldn't potentially show what was the issue. Adding as an image instead of a canvas I think makes the most sense.